pkg/virt-api/rest/subresource.go

```diff
@@ -974,28 +944,18 @@ func generateVMVolumeRequestPatch(vm *v1.VirtualMachine, volumeRequest *v1.Virtu
 		}
 	}
 
-	patchBytes, err := patch.GeneratePatchPayload(
-		patch.PatchOperation{
-			Op:    patch.PatchTestOp,
-			Path:  "/status/volumeRequests",
-			Value: vm.Status.VolumeRequests,
-		},
-		patch.PatchOperation{
-			Op:    getPatchVerb(vm.Status.VolumeRequests),
-			Path:  "/status/volumeRequests",
-			Value: vmCopy.Status.VolumeRequests,
-		},
+	patchSet := patch.New(
+		patch.WithTest("/status/volumeRequests", vm.Status.VolumeRequests),
 	)
 
-	return string(patchBytes), err
-}
-
-func getPatchVerb(requests []v1.VirtualMachineVolumeRequest) string {
-	verb := "add"
-	if len(requests) > 0 {
-		verb = "replace"
+	if len(vm.Status.VolumeRequests) > 0 {
+		patchSet.AddOption(patch.WithReplace("/status/volumeRequests", vmCopy.Status.VolumeRequests))
+	} else {
+		patchSet.AddOption(patch.WithAdd("/status/volumeRequests", vmCopy.Status.VolumeRequests))
 	}
-	return verb
+	patchBytes, err := patchSet.GeneratePayload()
+
+	return string(patchBytes), err
```

Comment:
Makes sense to change this func to return bytes too?

Reply:
Yes, done! Thanks